### ISSUE TITLE
В поле и свойстве "ЧИСЛО" возможность задавать несколько вариантов единиц измерения

### ISSUE THREAD

**Issue by vlad7090** [instantsoft/icms2#176]
Создаю каталог и, как уже писал, столкнулся с проблемой множественного выбора, но этот момент Fuze уже реализовал, **за что ему огромное спасибо.** Жаль нельзя к 2.4.0 прикрутить.

Но еще возникла проблема - возможность задавать несколько единиц измерения для свойства и поля ЧИСЛО. Зачем это надо и почему это полезно.

При создании поля, например цена, можно будет задавать единицы имерения рубли, доллары, евро и т.д. Либо например отчисления могут быть в рублях, а может указываться просто % от суммы. Таким образом можно было бы в поле задать значения "рубли и %".
Сейчас приходится все в рублях указывать, что в корне не верно, т.к продукция во всех валютах может закупаться , а цена валюты прыгает.
А так, создав поле ЧИСЛО и несколько вариантов единиц измерения, то можно будет задавать и указывать в какой валюте.

**Comment by vlad7090** [comment]
блин, странно, что никто не поддержал, хотя функция 100 очень полезна)

**Comment by pumpuic** [comment]
Полезная но достаточно таки специфична. Но на доске объявлений, например, была бы очень кстати.

**Comment by kdv1978** [comment]
Да, посмотрите встроенная геолокация включает в себя многие страны, а валюту задать стандартно можно только одну. На той же самой доске объявлений как пользователям задавать цену? либо универсально в usd, либо как пишет ТС возможность задавать несколько едениц измерения.

**Comment by Val-Git** [comment]
> Таким образом можно было бы в поле задать значения "рубли и %".

а что сейчас мешает так указать? Есть опция "Единица измерения" и указывайте в нее "рубли и %"

> геолокация включает в себя многие страны, а валюту задать стандартно можно только одну.

А в какой стране официально несколько валют?

**Comment by facepoint** [comment]
> а что сейчас мешает так указать? Есть опция "Единица измерения" и указывайте в нее "рубли и %"

Так вроде там нельзя через запятую указать варианты, чтобы потом при заполнении поля можно было выбрать одну из величин

**Comment by Val-Git** [comment]
> Так вроде там нельзя через запятую указать варианты, чтобы потом при заполнении поля можно было выбрать одну из величин

=) указать через запятую можно, а вот чтобы выбирало потом нельзя.
Но, мне теперь понятно чего народ хочет. А то как то пространно все отписываются))

**Comment by kdv1978** [comment]
"А в какой стране официально несколько валют?"
Так в том и проблема, если допустим доска объявлений стран СНГ, то и приходится брать за базу например $ т.к. нет возможности задать список валют в единицах измерения, вот и приходится всем вместо того что-бы каждому указать при подаче объявления цену в своей валюте, использовать то, что стоит в настройках (а задать можно одну валюту).

**Comment by Val-Git** [comment]
Легко и просто поле "апгрейднуть" не получиться. В БД оно записывается как int(11), а единицы измерения уже тянутся из настроек. Если задавать выбор валюты например, то этот выбор уже нужно хранить либо в отдельном поле БД либо менять тип поля с int на yaml_array и через набор костылей делать его разбор, фильтрацию выборку и др.
Первый вариант вроде как предпочтительней. Но это 2, в принципе, независимых поля, которые уже есть в составе iCMS2 поле число и поле список. Устанавливаете их вместе, стилями правите расположение в шаблоне и получаете что хотели! ))
